x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneSliceQueue.java

```diff
@@ -45,8 +46,8 @@ public int totalSlices() {
         return totalSlices;
     }
 
-    public Iterable<LuceneSlice> getSlices() {
-        return slices;
+    public Collection<String> remainingShards() {
```

Comment:
Should we name this method `remainingShardIds`, since it returns strings, not shards?

Reply:
I think that might be confused with `org.elasticsearch.index.shard.ShardId`. Probably this should be `remainingShardIdentifiers()` as it relies on `shardContext().shardIdentifier()`